### PR TITLE
Add support for Nuitka-Python to set link flags.

### PR DESCRIPTION
**Pull request by Maxwell175**
Thank your for contributing to Nuitka!

!! Please check that you select the **develop branch** (details see below link) !!

Before submitting a PR, please review the guidelines:
[Contributing Guidelines](https://github.com/kayhayen/Nuitka/blob/develop/CONTRIBUTING.md)

# What does this PR do?

Add support for Nuitka-Python to set link (and compile) flags.
Also fix an issue that caused Nuitka to not find dlls that python is linked to.

# Why was it initiated? Any relevant Issues?

# PR Checklist

- [x] Correct base branch selected? Should be `develop` branch.
- [x] Enabled commit hook or executed `./bin/autoformat-nuitka-source`.
- [ ] All tests still pass. Check the Developer Manual about `Running the Tests`.
      There are GitHub Actions tests that cover the most important
      things however, and you are welcome to rely on those, but they might not
      cover enough.
- [ ] Ideally new features or fixed regressions ought to be covered via new tests.
- [ ] Ideally new or changed features have documentation updates.

### CODE REVIEW

File: nuitka/freezer/DllDependenciesMacOS.py [hunk 1]
```diff
@@ -145,6 +146,7 @@ def _resolveBinaryPathDLLsMacOS(
 
     rpaths = _detectBinaryRPathsMacOS(original_dir, binary_filename)
     rpaths.update(package_specific_dirs)
+    print(rpaths)
```
Comment (kayhayen) [inline nuitka/freezer/DllDependenciesMacOS.py:149]: That stray print is going to kill tests

File: nuitka/freezer/DllDependenciesMacOS.py [hunk 2]
```diff
@@ -161,6 +163,14 @@ def _resolveBinaryPathDLLsMacOS(
         elif os.path.basename(path) == os.path.basename(binary_filename):
             # We ignore the references to itself coming from the library id.
             continue
+        elif not os.path.isabs(path) and not os.path.exists(path):
```
Comment (kayhayen) [inline nuitka/freezer/DllDependenciesMacOS.py:166]: So, relative paths with any '@' are a thing, where did you see these, how does macOS handle these?!

Comment (Maxwell175) [reply nuitka/freezer/DllDependenciesMacOS.py:166]: When setting up the linking of one of the necessary dll dependencies for one of the packages that i need for Nuitka-Python, that is the default way that it links it.

Here is the otool Load command:
```
Load command 27
          cmd LC_LOAD_DYLIB
      cmdsize 40
         name libCg.dylib (offset 24)
   time stamp 2 Wed Dec 31 19:00:02 1969
      current version 0.0.0
compatibility version 0.0.0
```

Here is where it searches:
```
dyld[39260]: Library not loaded: 'libCg.dylib'
  Referenced from: '/Users/maxwell/test/python3'
  Reason: tried: 'libCg.dylib' (no such file), '/usr/local/lib/libCg.dylib' (no such file), '/usr/lib/libCg.dylib' (no such file), '/Users/maxwell/test/libCg.dylib' (no such file), '/usr/local/lib/libCg.dylib' (no such file), '/usr/lib/libCg.dylib' (no such file)
```

File: nuitka/freezer/DllDependenciesMacOS.py (hunk 1 above)
Comment (Maxwell175) [reply nuitka/freezer/DllDependenciesMacOS.py]: Fixed.

File: nuitka/freezer/DllDependenciesMacOS.py (hunk 2 above)
Comment (Maxwell175) [reply nuitka/freezer/DllDependenciesMacOS.py]: It seems that this happens when a dll is linked in using the `-l` option and the library directory is specified with `-L`. This is really only an issue with Nuitka Python so I updated the code to be in a `isNuitkaPython` check and also to be more precise and robust.

**Pull request merged**